src/Data/List.purs

```diff
@@ -257,6 +258,13 @@ uncons :: forall a. List a -> Maybe { head :: a, tail :: List a }
 uncons Nil = Nothing
 uncons (x : xs) = Just { head: x, tail: xs }
 
+-- | Break a list into its last element, and the preceding elements,
+-- | or `Nothing` if the list is empty.
+-- |
+-- | Running time: `O(2n)`
+unsnoc :: forall a. List a -> Maybe { init :: List a, last :: a }
+unsnoc xs = { init: _, last: _ } <$> init xs <*> last xs
```

Comment:
Actually, can we do this in a single pass? Otherwise, the user doesn't gain anything over using the functions separately.